src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -493,5 +613,259 @@ public async Task<ContentItem> CloneAsync(ContentItem contentItem)
+            }
+            else if (contentItem.Published)
+            {
+                // When importing a published item existing drafts and existing published must be removed.
+                // Otherwise an existing draft would become an orphan and if published would overwrite
+                // the imported (which we assume is the version that wins) content.
+                await RemoveVersionsAsync(contentItem, evictionVersions);
+            }
+            // When neither imported or latest the operation will create a database record
+            // which will be part of the content item archive.
+
+            // Invoked create handlers.
+            var context = new CreateContentContext(contentItem);
+            await Handlers.InvokeAsync((handler, context) => handler.CreatingAsync(context), context, _logger);
+            await ReversedHandlers.InvokeAsync((handler, context) => handler.CreatedAsync(context), context, _logger);
+
+            // The content item should be placed in the session store so that further calls
+            // to ContentManager.Get by a scoped index provider will resolve the imported item correctly.
+            _session.Save(contentItem);
+            _contentManagerSession.Store(contentItem);
+
+            await UpdateAsync(contentItem);
+
+            var result = await ValidateAsync(contentItem);
+            if (!result.Succeeded)
+            {
+                return result;
+            }
+
+            if (contentItem.Published)
+            {
+                // Invoke published handlers to add information to persistent stores
+                var publishContext = new PublishContentContext(contentItem, null);
+
+                await Handlers.InvokeAsync((handler, context) => handler.PublishingAsync(context), publishContext, _logger);
+                await ReversedHandlers.InvokeAsync((handler, context) => handler.PublishedAsync(context), publishContext, _logger);
+            }
+
+            // Restore values that may have been altered by handlers.
+            if (modifiedUtc.HasValue)
+            {
+                contentItem.ModifiedUtc = modifiedUtc;
+            }
+            if (publishedUtc.HasValue)
+            {
+                contentItem.PublishedUtc = publishedUtc;
+            }
+
+            // There is a risk here that the owner or author does not exist in the importing system.
+            // We check that at least a value has been supplied, if not the owner property and author
+            // property would be left as the user who has run this import.
+            if (!String.IsNullOrEmpty(owner))
+            {
+                contentItem.Owner = owner;
+            }
+            if (!String.IsNullOrEmpty(author))
+            {
+                contentItem.Author = author;
+            }
+
+            // Re-enlist content item here in case of session queries.
+            _session.Save(contentItem);
```

Comment:
I'm afraid that when importing, e.g if more than one batch, in the next loop another session query will trigger an auto flush and then clear the session saved list. Or as suggested above, and if possible, create a scope per batch.

Maybe we need to do the save through another service that does a session save immediately and then do it again at the end of the shell scope, as i did in my branch for committing though an `IDocumentStore`, see #6008

Reply:
@jtkech I will have to check this further, but has some behaviour in YesSql changed around `AutoFlush()` ?

I see a lot of comments about re-enlisting the save in our code, but when I look in the YesSql code, `FlushAsync` always Saves the entities, before clearing the updated / saved lists? 
But it does not commit the transaction, until it is disposed

My testing seems to confirm this, but as I say, there are a lot of comments about re-enlisting for save in our code.

The only reason I use the batching here, is for the version queries, to avoid this issue https://github.com/sebastienros/yessql/issues/187

So yes, if we need to we could a session shell scope, though I was preferring to be able to do it all as one session, to be able to cancel the entire import, if one of the validations fail. 

I will adapt one of the tests to run multiple batches (but not commit it, as it would slow the CI) to check the behaviour here